assets/src/edit-story/components/panels/textStyle/textStyle.js

```diff
@@ -82,23 +95,42 @@ function StylePanel({ selectedElements, pushUpdate }) {
     },
   } = useRichTextFormatting(selectedElements, pushUpdate);
 
+  const setLetterSpacingMinMax = useCallback(
+    (value) => handleSetLetterSpacing(clamp(value, MIN_MAX.LETTER_SPACING)),
+    [handleSetLetterSpacing]
+  );
+
+  usePresubmitHandler(({ lineHeight: newLineHeight }) => {
+    return {
+      lineHeight: clamp(newLineHeight, MIN_MAX.LINE_HEIGHT),
+    };
+  }, []);
+
   return (
     <>
       <Row>
         <ExpandedNumeric
           aria-label={__('Line-height', 'web-stories')}
           float={true}
           value={lineHeight || 0}
+          min={MIN_MAX.LINE_HEIGHT.MIN}
+          max={MIN_MAX.LINE_HEIGHT.MAX}
           suffix={<OffsetVertical />}
           onChange={(value) => pushUpdate({ lineHeight: value })}
+          canBeEmpty
+          canBeNegative
```

Comment:
@barklund Even if a negative value can be inserted then it still doesn't actually allow negative values due to the min-max attributs set [based on the doc](https://docs.google.com/spreadsheets/d/1hdru-_p4Yk-VFQ1rEFGYHHI2A7eNZkxLA33-LblG8sE/edit#gid=0) (e.g. -2 will be converted to 0.5 on blurring). Also, I can start typing negative values but not when starting with `0`, e.g. `-0.5` can't be typed in.

Perhaps it would be good to confirm the negative values for line-height etc. since UAT will probably be done based on the doc?

Reply:
> Perhaps it would be good to confirm the negative values for line-height etc. since UAT will probably be done based on the doc?

Agree, there is no negative value allowed right now (min/max spec), though it might be useful for rotation.

Reply:
Ah, that makes sense. Will confirm with @o-fernandez.

Reply:
Playing with it, it actually doesn't make sense to set either to negative (maybe small negative values for letter spacing, but it's a future optimization). Removed it again.